New: Add grouped-accessor-pairs rule (fixes #12277) (#12331)
eslint/eslint: docs/rules/grouped-accessor-pairs.md

# Require grouped accessor pairs in object literals and classes (grouped-accessor-pairs)

> Getter and setter for the same property don't necessarily have to be defined one after the other.

Reviewer: ```suggestion
A getter and setter for the same property don't necessarily have to be defined adjacent to each other.
```

Author: Thanks! Had trouble with the opening sentence

        this.val = value;
    },
    get a() {
        return this.val;
    }
};

class Foo {
    set a(value) {
        this.val = value;
    }
    get a() {
        return this.val;
    }
}

const Bar = class {
    static set a(value) {
        this.val = value;
    }
    static get a() {
        return this.val;
    }
}
```

## Known Limitations

Due to the limits of static analysis, this rule does not account for possible side effects and in certain cases
might require or miss to require grouping or order for getters/setters that have a computed key, like in the following example:

```js
/*eslint grouped-accessor-pairs: "error"*/

var a = 1;

// false warning (false positive)
var foo = {
    get [a++]() {
        return this.val;
    },
    b: 1,
    set [a++](value) {
        this.val = value;
    }
};

// missed warning (false negative)
var bar = {
    get [++a]() {
        return this.val;
    },
    b: 1,
    set [a](value) {
        this.val = value;
    }
};
```

Also, this rule does not report any warnings for properties that have duplicate getters or setters.

See [no-dupe-keys](no-dupe-keys.md) if you also want to disallow duplicate keys in object literals.

> See [no-dupe-class-members](no-dupe-class-members.md) if you also want to disallow duplicate names in class definitions.

Author: Done! :-)
